libkbfs/block_retrieval_worker.go

```diff
@@ -48,12 +48,12 @@ func newBlockRetrievalWorker(bg blockGetter,
 // blockGetter.getBlock, and responds to the subscribed requestors with the
 // results.
 func (brw *blockRetrievalWorker) HandleRequest() (err error) {
-	brw.queue.Work(brw.workCh)
 	var retrieval *blockRetrieval
 	select {
-	case retrieval = <-brw.workCh:
+	case <-brw.workCh:
+		retrieval = brw.queue.popIfNotEmpty()
```

Comment:
I think that's okay as long as there's a quid-pro-quo. I work on your request and you'll work on mine. The same number of workers is allocated on both on-demand requests and prefetch requests, so no starvation should be introduced.